airbyte-webapp/src/hooks/services/useConnectionHook.tsx

```diff
@@ -77,12 +77,14 @@ export const useConnectionLoad = (
   connectionId: string
 ): {
   connection: WebBackendConnectionRead;
-  refreshConnectionCatalog: () => Promise<WebBackendConnectionRead>;
+  refreshConnectionCatalog: () => Promise<void>;
 } => {
-  const connection = useGetConnection(connectionId);
+  const [connection, setConnection] = useState(useGetConnection(connectionId));
   const connectionService = useWebConnectionService();
 
-  const refreshConnectionCatalog = async () => await connectionService.getConnection(connectionId, true);
+  const refreshConnectionCatalog = async () => {
+    setConnection(await connectionService.getConnection(connectionId, true));
+  };
```

Comment:
Refreshing the connection now sets the initial connection object instead of providing a new one.